release-24.1: bootstrap: unify tenant and sys tables

dt opened the pull request:
Backport 2/2 commits from #122005.

/cc @cockroachdb/release

---

We want to be able to replicate a system tenant into a non-system tenant
but this requires that they have the same set of system tables so that
when the non-system tenant starts executing, it finds all the same tables
that it expects to exist, regardless of whether it was created via PCR of
a system tenant or bootstrap of a secondary tenant.


Release note: none.
Epic: none.

Release justification: small (tests notwithstanding) change we need in a major version before we can enable/fix streaming from system tenants.

dt committed:
bootstrap: unify tenant and sys tables

We want to be able to replicate a system tenant into a non-system tenant
but this requires that they have the same set of system tables so that
when the non-system tenant starts executing, it finds all the same tables
that it expects to exist, regardless of whether it was created via PCR of
a system tenant or bootstrap of a secondary tenant.

Release note: none.
Epic: none.

dt committed:
upgrades: add the span_count table to sys tenants

They need it if they will be replicated via PCR later so we create it
during the upgrade if it is missing. It was previously added to bootstrap.

Epic: none.
Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/122123)
<!-- Reviewable:end -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

I think for the purpose of your usage here it doesn't matter, but should we also longer term have a migration to create any other missing tables?

> Reviewed 100 of 100 files at r1, 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/122123#-:-NvCc99d1kt_ffhGOmql:b-dh1iqb)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dt, @msbutler, and @stevendanna)

___
*[`pkg/upgrade/upgrades/v24_1_add_span_counts_test.go` line 52 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/122123#-NvCc5-v39hhTqGW2DjC:-NvCc5-v39hhTqGW2DjD:b6u24j8) ([raw file](https://github.com/cockroachdb/cockroach/blob/d546a4e9b693c9c0016ace89e5ec62ab2008aea1/pkg/upgrade/upgrades/v24_1_add_span_counts_test.go#L52)):*
> ```Go
> 	require.True(t, s.ExecutorConfig().(sql.ExecutorConfig).Codec.ForSystemTenant())
> 	upgrades.Upgrade(t, sqlDB, clusterversion.V24_1_AddSpanCounts, nil, false)
> 	_, err := sqlDB.Exec("SELECT * FROM system.public.span_count")
> ```

Nit: Probably a good idea to confirm it didn't exist before


<!-- Sent from Reviewable.io -->

dt committed:
cli: fix merge skew

https://github.com/cockroachdb/cockroach/pull/121989 and  https://github.com/cockroachdb/cockroach/pull/121290

Release note: none.
Epic: none.
